packages/main/src/plugin/extension-loader.spec.ts

```diff
@@ -803,3 +819,458 @@ test('Verify extension uri', async () => {
   expect(grabUri).toBeDefined();
   expect(grabUri.fsPath).toBe('dummy');
 });
+
+describe('Navigation', async () => {
+  test('navigateToContainers', async () => {
+    const api = extensionLoader.createApi('path', {
+      name: 'name',
+      publisher: 'publisher',
+      version: '1',
+      displayName: 'dname',
+    });
+
+    // Spy send method
+    const sendMock = vi.spyOn(apiSender, 'send');
+
+    await api.navigation.navigateToContainers();
+    expect(sendMock).toBeCalledWith('navigate', { page: NavigationPage.CONTAINERS });
+  });
+  test('navigateToContainerX existing container', async () => {
+    const api = extensionLoader.createApi('path', {
+      name: 'name',
+      publisher: 'publisher',
+      version: '1',
+      displayName: 'dname',
+    });
+
+    // Mock listSimpleContainer implementation
+    const listSimpleContainerSpy = vi.spyOn(containerProviderRegistry, 'listSimpleContainers');
+    listSimpleContainerSpy.mockImplementation(async () => [
+      { Id: 'valid' } as unknown as containerDesktopAPI.ContainerInfo,
+    ]);
+    // Spy send method
+    const sendMock = vi.spyOn(apiSender, 'send');
+
+    // First let's define our test cases
+    const tests: {
+      name: string;
+      method: (id: string) => Promise<void>;
+      expected: NavigationRequest;
+    }[] = [
+      {
+        name: 'navigateToContainer',
+        method: api.navigation.navigateToContainer,
+        expected: {
+          page: NavigationPage.CONTAINER,
+          parameters: {
+            id: 'valid',
+          },
+        },
+      },
+      {
+        name: 'navigateToContainerLogs',
+        method: api.navigation.navigateToContainerLogs,
+        expected: {
+          page: NavigationPage.CONTAINER_LOGS,
+          parameters: {
+            id: 'valid',
+          },
+        },
+      },
+      {
+        name: 'navigateToContainerInspect',
+        method: api.navigation.navigateToContainerInspect,
+        expected: {
+          page: NavigationPage.CONTAINER_INSPECT,
+          parameters: {
+            id: 'valid',
+          },
+        },
+      },
+      {
+        name: 'navigateToContainerTerminal',
+        method: api.navigation.navigateToContainerTerminal,
+        expected: {
+          page: NavigationPage.CONTAINER_TERMINAL,
+          parameters: {
+            id: 'valid',
+          },
+        },
+      },
+    ];
+
+    for (const testCase of tests) {
+      // Call the method provided
+      await testCase.method('valid');
```

Comment:
I think you should use a parameterized test that would better fit in the test model

with test.each() / https://vitest.dev/api/#test-each

example in Podman Desktop https://github.com/containers/podman-desktop/blob/4767272a1562e9b4cfc85674578228abccfbfd63/packages/main/src/plugin/types/uri.spec.ts#L63-L69